### DISCUSSION TITLE
Use Form.Item with Checkbox.Group

### DISCUSSION THREAD

**Discussion by bombillazo** [ant-design/ant-design#42531] [Q&A] [answered]
Hello. I am attempting to get the values of a Checkbox.Group inside a Form.Item, but I need help. 

```tsx
<Form.Item
  name="items
  valuePropName="value"
>
  <Checkbox.Group value={value}>
     {_options.map((option, idx) => (
        <Checkbox
          key={idx}
          value={option.value}
          onChange={(e) => {
            const isChecked = e.target.checked;
            if (isChecked) setValue([...value, option.value]);
            else
              setValue(
                value.filter((v) => {
                  return v !== option.value;
                }),
              );
          }}>
          {option.label}
        </Checkbox>
      ))}
  </ Checkbox.Group>
</ Form.Item>
```

This seems not to work as expected since `value` is updated but inside the `onFinish` function of the form I get an empty array...

I need to use this way vs the `options` prop because I have additional custom logic I need to render inside some <Checkbox> components.

**Comment by BoyYangzai** [comment]
@bombillazo 
It seems work fine
try this demo
https://codesandbox.io/s/fu-za-yi-dian-de-kong-jian-antd-5-5-1-forked-3g62cw?file=/demo.tsx

**Reply by bombillazo** [reply]
Hey @BoyYangzai Thanks for the demo.

I am trying to do the following:

<img width="403" alt="image" src="https://github.com/ant-design/ant-design/assets/44002321/ca5cd6aa-7ea2-4d55-b35d-42bba4023204">

Have an additional checkbox that can be used to input a custom value, but that input value is not appearing in teh submit when teh input changes.

https://codesandbox.io/s/fu-za-yi-dian-de-kong-jian-antd-5-5-1-forked-65h1u1?file=/demo.tsx

**Reply by BoyYangzai** [reply]
@bombillazo 
hi bro😄
i update your demo ，it works fine 
https://codesandbox.io/s/fu-za-yi-dian-de-kong-jian-antd-5-5-1-forked-md4kdo

**Reply by bombillazo** [reply]
Hmmmmm If I change the text input value, the checkbox is deselected :/ I want eh checkbox to stay selected and the value to change so when I click submit, the latest value is sent. RIght not it only sends the last value when the checkbox was click to true.

**Reply by bombillazo** [reply, marked as answer]
Nevermind! I ended up creating a custom input object that can handle the `value` and `onChange` with Form.Item 😃

**Reply by phantrung18072001** [reply]
@bombillazo I have the same problem @@, how to make checkbox not deselected after input text. I need your help, thanks
